cyclonedx/model/tool.py

```diff
@@ -177,7 +177,7 @@ def __bool__(self) -> bool:
         return any([self._tools, self._components, self._services])
 
     @property
-    def components(self) -> SortedSet[Component]:
+    def components(self) -> 'SortedSet[Component]':
```

Comment:
using strings instead of actual classes will hinder the serialization lib from working properly.

Reply:
found the tests. all good

Reply:
It seems to work OK, but it does expose an interesting bug. In the Tool class, if I leave the quotes on
```py
    def hashes(self) -> 'SortedSet[HashType]':
```
and
```py
    def external_references(self) -> 'SortedSet[ExternalReference]':
```
tests pass. However, if I remove the quotes, tests fail with this (snipped for just the failure).

For hahes:
```
        <tool>
          <vendor>CycloneDX</vendor>
          <name>cyclonedx-python-lib</name>
          <version>TESTING</version>
-         <hashes/>
        </tool>
```
(It removes the empty hashes when the quotes aren't there)
And for external references, it renders it like so:
```
        <externalReferences>SortedSet([&lt;ExternalReference BUILD_SYSTEM, https://github.com/CycloneDX/cyclonedx-python-lib/actions&gt;, &lt;ExternalReference DISTRIBUTION, https://pypi.org/project/cyclonedx-python-lib/&gt;, &lt;ExternalReference DOCUMENTATION, https://cyclonedx-python-library.readthedocs.io/&gt;, &lt;ExternalReference ISSUE_TRACKER, https://github.com/CycloneDX/cyclonedx-python-lib/issues&gt;, &lt;ExternalReference LICENSE, https://github.com/CycloneDX/cyclonedx-python-lib/blob/main/LICENSE&gt;, &lt;ExternalReference RELEASE_NOTES, https://github.com/CycloneDX/cyclonedx-python-lib/blob/main/CHANGELOG.md&gt;, &lt;ExternalReference SCM, https://github.com/CycloneDX/cyclonedx-python-lib&gt;, &lt;ExternalReference WEBSITE, https://github.com/CycloneDX/cyclonedx-python-lib/#readme&gt;])</externalReferences
```
and also produces this error:
```
AssertionError: <string>:11:0:ERROR:SCHEMASV:SCHEMAV_CVC_COMPLEX_TYPE_2_3: Element '{http://cyclonedx.org/schema/bom/1.4}externalReferences': Character content other than whitespace is not allowed because the content type is 'element-only'. is not None
```
So, it seems those are bugs, but all works fine when the return type is quoted, so I'm not sure what's going on there.

In the case of ToolRepository, the tests pass without or without the return types quoted.

Reply:
Oh! I remember why I quoted those return types: in Python 3.8, SortedSet isn't subscriptable because the classes it inherits from aren't subscriptable.